### PR TITLE
EEH for block devices are updated. Added validation to check multipath devices

### PR DESCRIPTION
**Pull request by maramsmurthy**
… devices

after EEH.

EEH for block devies are updated. Added validation to check multipath devices after EEH.

Signed-off-by: Maram Srimannarayana Murthy <Maram.Srimannarayana.Murthy@ibm.com>

### CODE REVIEW

**Comment by maramsmurthy** [comment]
Raised new request due to branch issue with earlier pull request

**Comment by maramsmurthy** [comment]
Addressed all review comments mentioned in 
https://github.com/avocado-framework-tests/avocado-misc-tests/pull/2478

File: io/pci/EEH.py [hunk 1]
```diff
@@ -134,6 +158,22 @@ def test_eeh_basic_pe(self):
...
+                        self.log.info("Paths after EEH: %s",
+                                      paths_after_eeh)
+                        self.log.info("hosts state after EEH: %s",
+                                      host_state_after_eeh)
+                        if paths_before_eeh != paths_after_eeh:
+                            self.fail("All Paths are failed to recover"
+                                      " after EEH")
+                        if host_state_before_eeh != host_state_after_eeh:
+                            self.fail("Hosts state differ after EEH")
+
```
Comment (Naresh-ibm) [inline io/pci/EEH.py:176]: @maramsmurthy where are we checking the states of the paths here.
that is the major functionality check here what we need to incorporate.
follow below steps while doing validation,
1. avoid static sleep times, and use wait_for()
2.  after each eeh hit, you should check each paths_status== online, and fc_host_state == online.

what if you have the adapter already in faulty state?
what if if you have some eeh hit before u try to inject error and few paths are already in failed state?
you will be still working on a fault adapter and making the test case PASS.

Comment (Naresh-ibm) [reply io/pci/EEH.py:176]: please refer, port_bounce or pci_hotplug test script.
or let me know I can help you out to add the validation part.

**Review by Naresh-ibm — CHANGES_REQUESTED** [review]

File: io/pci/EEH.py [hunk 2]
```diff
@@ -63,7 +64,22 @@ def setUp(self):
...
+            dmesg_op = process.system_output('dmesg',
+                                             ignore_status=True,
+                                             shell=True).decode("utf-8")
+            if 'times in the last hour' in dmesg_op:
+                eeh_already_hit = re.findall(r"This PCI device has failed "
+                                             "([0-9]+) times in the last hour and will be "
+                                             "permanently disabled after ([0-9]+) failures", dmesg_op)
+                self.log.info('max_freeze value is adjusted as %s eeh hits'
+                              ' already observed', eeh_already_hit[-1][0])
+                self.max_freeze -= int(eeh_already_hit[-1][0])
```
Comment (abdhaleegit) [inline io/pci/EEH.py:82]: If the adapter already disabled due to max hits. than why adjusting value.. or if eeh not yet trigger there is no point checking dmesg .. am I missing something here ?

File: io/pci/EEH.py [hunk 3]
```diff
@@ -134,6 +158,22 @@ def test_eeh_basic_pe(self):
                                 break
                             else:
                                 self.log.info("PE recovered successfully")
+                    time.sleep(10)
+                    if (self.pci_class_name == 'scsi_host' and
```
Comment (abdhaleegit) [inline io/pci/EEH.py:162]: so no validation for other than scsi_host classes ?

**Review by abdhaleegit — CHANGES_REQUESTED** [review]

**Review by abdhaleegit — CHANGES_REQUESTED** [review]
Can you  please reformat the commit message

```
EEH for block devies are updated. Added validation to check multipath…
… devices

after EEH.

EEH for block devies are updated. Added validation to check multipath devices
after EEH.
```

**Comment by abdhaleegit** [comment]
@maramsmurthy Any update please ?

**Comment by abdhaleegit** [comment]
@maramsmurthy Please push the latest code

File: io/pci/EEH.py [hunk 4]
```diff
@@ -63,7 +64,22 @@ def setUp(self):
         eeh_enable_file = "/sys/kernel/debug/powerpc/eeh_enable"
         if '0x1' not in genio.read_file(eeh_enable_file).strip():
             self.cancel("EEH is not enabled, please enable via FSP")
-        self.max_freeze = self.params.get('max_freeze', default=1)
+        self.max_freeze = self.params.get('max_freeze')
```
[the remark this answers is not in the document]
Comment (maramsmurthy) [reply io/pci/EEH.py:67]: Raised new PR "https://github.com/avocado-framework-tests/avocado-misc-tests/pull/2590". It's addressed there.

File: io/pci/EEH.py (hunk 2 above)
Comment (maramsmurthy) [reply io/pci/EEH.py:82]: Will be addressing it in PR "https://github.com/avocado-framework-tests/avocado-misc-tests/pull/2590"

File: io/pci/EEH.py (hunk 3 above)
Comment (maramsmurthy) [reply io/pci/EEH.py:162]: Will be addressing it in PR "https://github.com/avocado-framework-tests/avocado-misc-tests/pull/2590"

File: io/pci/EEH.py (hunk 1 above)
Comment (maramsmurthy) [reply io/pci/EEH.py:176]: Addressing it in PR "https://github.com/avocado-framework-tests/avocado-misc-tests/pull/2590". New validation added for every path.

**Pull request closed without merging**